packages/contracts/contracts/L1/AssetManager.sol

```diff
@@ -816,6 +816,81 @@ contract AssetManager is ISemver, IERC721Receiver, IAssetManager {
         }
     }
 
+    /**
+     * @inheritdoc IAssetManager
+     */
+    function claimKghReward(address validator, uint256[] calldata tokenIds) external {
+        if (validator == address(0)) revert ZeroAddress();
+        if (tokenIds.length == 0) revert NotAllowedZeroInput();
```

Comment:
I think it's better to allow only the case to input all the tokenIds the delegator has delegated. Otherwise, we need to allow `tokenIds.length == 0` case if the delegator wants to claim only the boosted reward. I think it's better to restrict the case to simplify. Or it also can be a way to add `tokenIds` array in `KghDelegator`, not to input `tokenIds` in this function.

Reply:
Why do we need to check `UNDELEGATION_DELAY_SECONDS` here? I thought since they are not withdrawing their KGH, it was not necessary.

Regarding the second comment, I think it depends on how frontend is implemented. I suppoesed that there is a list of KGH tokens shown there, so users can pick what KGHs they wanna claim rewards at or just claim all. If there's just only one button with 'Claim all rewards', then maybe you're right.

The last point will increase gas fees in delegating and undelegating KGHs, so I didn't see it's desirable.

Reply:
I also think it is not withdrawal, we don't need to check this. But I would be good to have view func that how much is the claimable reward including both base reward and boosted reward.

Reply:
That makes sense. It's unnecessary to check `UNDELEGATION_DELAY_SECONDS` here.

Hmm then let's clear the cases.
1. claim only boosted reward (tokenIds.length == 0)
2. claim boosted reward and part of base reward (tokenIds.length < kghDelegator.kghNum)
3. claim boosted reward and all of base reward (tokenIds.length == kghDelegator.kghNum)

 Also, I think we need to check if the token Id is really delegated by msg.sender or not using kghDelegator.delegatedAt. Only checking the length of tokenIds array could make side effects.

Reply:
> I also think it is not withdrawal, we don't need to check this. But I would be good to have view func that how much is the claimable reward including both base reward and boosted reward.

I have added `getKghReward` function in the interface. It would be nice to implement it in this PR.

Reply:
Got it, then I think sticking to the case for claiming all rewards (base, boosted) will be better. If we cannot claim  portion of (base + boosted), making users claiming only boosted reward or all boosted reward + portion of base reward is not meaningful that much. I was confused at the above part of claiming portion of boosted reward.

The remaining parts you suggested look all good, so I'll add the check and `getKghReward` function.

Reply:
OK, then let's confirm if all the tokenIds are provided as input, and they are really delegated by the `msg.sender`.

Reply:
> Also, I think we need to check if the token Id is really delegated by msg.sender using kghDelegator.delegatedAt. Only checking the length of tokenIds array could make side effects.

valid point.